nodewatcher/modules/monitor/http/survey/processors.py

```diff
@@ -0,0 +1,92 @@
+import datetime
+import pytz
+
+from django.utils.translation import gettext_noop
+
+from nodewatcher.core.monitor import models as monitor_models, processors as monitor_processors
+from nodewatcher.utils import ipaddr
+from nodewatcher.modules.monitor.sources.http import processors as http_processors
+
+from django_datastream import datastream
+
+DATASTREAM_SUPPORTED = False
+try:
+    from nodewatcher.modules.monitor.datastream import fields as ds_fields, models as ds_models, base as ds_base
+    from nodewatcher.modules.monitor.datastream.pool import pool as ds_pool
+
+    class SurveyInfoStreams(ds_base.StreamsBase):
+        neighbor_graph = ds_fields.GraphField(tags={
+            'title': gettext_noop("Neighbor topology"),
+            'description': gettext_noop("Neighbor topology."),
+            'visualization': {
+                'type': 'graph',
+                'initial_set': True,
+            }
+        })
+
+        def get_module_name(self):
+            return 'monitor.http.survey'
+
+        def get_stream_query_tags(self):
+            return {'module': 'survey'}
+
+        def get_stream_tags(self):
+            return {'module': 'survey'}
+
+        def get_stream_highest_granularity(self):
+            return datastream.Granularity.Minutes
+
+    class SurveyInfoStreamsData(object):
+        def __init__(self, vertices, edges):
+            self.neighbor_graph = {
+                'v': vertices,
+                'e': edges
+            }
+
+    ds_pool.register(SurveyInfoStreamsData, SurveyInfoStreams)
+
+    DATASTREAM_SUPPORTED = True
+except ImportError:
+    pass
+
+
+class SurveyInfo(monitor_processors.NodeProcessor):
+    """
+    Stores neighbor's reported channel, BSSID and signal strength into the monitoring schema. A star shaped graph
```

Comment:
Are you really storing anything into the monitoring schema? I thin you are storing just into datastream?

In this case you might want to get rid of whole `DATASTREAM_SUPPORTED` stuff and simply require datastream for this module. Like topology requires it. Because the point of `DATASTREAM_SUPPORTED` is that you can store something only into monitoring schema, but not into time-series database. So that you have only latest values.

But you are not storing latest values, because monitoring schema does not support graph types anyway.